// @ampt/sdk/dist/params.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// params/index.ts
var params_exports = {};
__export(params_exports, {
  Params: () => Params,
  params: () => params
});
module.exports = __toCommonJS(params_exports);
var existingEnv = {
  ORG_NAME: process.env.AMPT_ORG_NAME,
  SERVICE_NAME: process.env.AMPT_SERVICE_NAME,
  INSTANCE_NAME: process.env.AMPT_INSTANCE_NAME,
  INSTANCE_TYPE: process.env.AMPT_INSTANCE_TYPE,
  APP_NAME: process.env.AMPT_SERVICE_NAME,
  STAGE_NAME: process.env.AMPT_STAGE,
  AMPT_URL: process.env.AMPT_URL,
  AMPT_REGION: process.env.AMPT_REGION
};
var Params = class {
  params;
  constructor() {
    this.params = {
      ...JSON.parse(process.env.AMPT_PARAMS || "{}") || {},
      ...JSON.parse(process.env.PARAMS || "{}") || {},
      ...existingEnv
    };
  }
  export(keys = []) {
    if (!keys.length) {
      process.env = {
        ...this.params,
        ...process.env
      };
    } else {
      keys.forEach((key) => {
        if (!process.env[key]) {
          process.env[key] = this.params[key];
        }
      });
    }
  }
  list() {
    return this.params;
  }
  handler(key) {
    if (key) {
      return this.params[key] || void 0;
    }
    return {
      export: this.export.bind(this),
      list: this.list.bind(this)
    };
  }
};
var _params = new Params();
var params = _params.handler.bind(_params);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Params,
  params
});
